Fix `Build Docker image` step in `publish.yaml` workflow

* There is an issue with `publish.yaml` workflow where it says:
* `ERROR: failed to build: invalid tag "ghcr.io/Askie01/recipe-application:1.0.0": repository name must be lowercase Error: Process completed with exit code 1.`

* The issue seems to occur on `Build Docker image` step in the mentioned workflow.

--- a/.github/workflows/publish.yaml
+++ b/.github/workflows/publish.yaml
@@ -49,10 +49,13 @@ jobs:
 
       - name: Build Docker image
         run: |
+          REPO_LOWER=$(echo "${GITHUB_REPOSITORY}" | tr '[:upper:]' '[:lower:]')
           docker build \
-           -t ghcr.io/${{ github.repository }}:${{ inputs.version }} \
-           -t ghcr.io/${{ github.repository }}:latest \
-           .
+            -t ghcr.io/$REPO_LOWER:${{ inputs.version }} \
+            -t ghcr.io/$REPO_LOWER:latest \
+            .
+        env:
+          GITHUB_REPOSITORY: ${{ github.repository }}
 
       - name: Push Docker image
         run: |

